benchmarks/statevec.py

```diff
@@ -0,0 +1,208 @@
+        if backend.pattern.seq[-1] == cmd:
+            backend.finalize()
+
+
+# %%
+# Then, we define a random circuit generator:
+
+
+def simple_random_circuit(nqubit, depth):
+    r"""Generate a random circuit.
+
+    This function generates a circuit with nqubit qubits and depth layers, having layers of CNOT and Rz gates with random placements.
+
+    Parameters
+    ----------
+    nqubit : int
+        number of qubits
+    depth : int
+        number of layers
+
+    Returns
+    -------
+    circuit : graphix.transpiler.Circuit object
+        generated circuit
+    """
+    qubit_index = [i for i in range(nqubit)]
+    circuit = Circuit(nqubit)
+    for _ in range(depth):
+        np.random.shuffle(qubit_index)
+        for j in range(len(qubit_index) // 2):
+            circuit.cnot(qubit_index[2 * j], qubit_index[2 * j + 1])
+        for j in range(len(qubit_index)):
+            circuit.rz(qubit_index[j], 2 * np.pi * np.random.random())
+    return circuit
+
+
+# %%
+# We define the test cases: shallow (depth=1) random circuits, only changing the number of qubits.
+
+test_cases = [(i, 1) for i in range(2, 24)]
+
+pattern_time = []
+circuit_time = []
+
+# %%
+# We then run simulations.
+# First, we run the pattern simulations and circuit simulation.
+
+for nqubit, depth in test_cases:
+    circuit = simple_random_circuit(nqubit, depth)
+    pattern = circuit.transpile()
+    pattern.standardize()
+    pattern.minimize_space()
+    max_qubit_num = 20 if nqubit < 20 else 50
+    backend = StatevectorBackend(pattern, max_qubit_num=max_qubit_num)
+    print(f"max space for nqubit={nqubit} circuit is ", pattern.max_space())
+    start = perf_counter()
+    run(backend)
+    end = perf_counter()
+    print(f"nqubit: {nqubit}, depth: {depth}, time: {end - start}")
+    pattern_time.append(end - start)
+    start = perf_counter()
+    circuit.simulate_statevector()
+    end = perf_counter()
+    circuit_time.append(end - start)
+
+
+# %%
+# Here we take benchmarking for MBQC simulation using `paddle_quantum`.
+
+
+def simple_random_circuit_for_paddle_quantum(width, depth):
+    r"""Generate a random circuit for paddle.
+
+    This function generates a circuit with nqubit qubits and depth layers, having layers of CNOT and Rz gates with random placements.
+
+    Parameters
+    ----------
+    width : int
+        number of qubits
+    depth : int
+        number of layers
+
+    Returns
+    -------
+    circuit : paddle_quantum.mbqc.qobject.Circuit
+        generated circuit
+    """
+    qubit_index = [i for i in range(width)]
+    circuit = PaddleCircuit(width)
+    for _ in range(depth):
+        np.random.shuffle(qubit_index)
+        for j in range(len(qubit_index) // 2):
+            circuit.cnot(which_qubits=[qubit_index[2 * j], qubit_index[2 * j + 1]])
+        for j in range(len(qubit_index)):
+            circuit.rz(which_qubit=qubit_index[j], theta=to_tensor(2 * np.pi * np.random.random(), dtype="float64"))
```

Comment:
instead of generating randomly for each packages, we should simulate for exact same circuits. will update the code.